### ISSUE TITLE
SA1402 does not fire for record classes

### ISSUE THREAD

**Issue by jberezanski** [DotNetAnalyzers/StyleCopAnalyzers#3542]
Repro:

```csharp
internal class Class1
{
}

internal class Class2 // SA1402 File may only contain a single type (expected)
{
}

internal record Class3 // no warning
{
}

internal record class Class4 // no warning
{
}
```

Expected behavior: SA1402 is reported for record classes, same as for non-record classes.

C# version: 10.0
StyleCopAnalyzers: 1.2.0-beta.406, 1.2.0-beta.435

**Comment by jberezanski** [comment]
In the product I'm working on we write domain models using immutable types. Record classes help us by eliminating some boilerplate, but most of our `record class` types are full, rich objects, encapsulating behavior and data (as per the Domain Model pattern), so definitely not dumb data containers. Those types should definitely be in separate files (which is why I created this issue in the first place).

In other words, I view `record` as being a _modifier/subcategory_ of the `class` and `struct` categories, not a category on its own (this is, by the way, reflected in the language syntax), and I would like for SA1402 to act on record classes and structs by default in the same manner as on non-record types of those categories (trigger for classes, ignore structs).

I understand, however, that record classes may often be used indeed as "dumb data classes" and people might want to keep them in the same files, so the rule behavior for records would probably need to be configurable.

**Comment by sharwell** [comment]
I would expect StyleCop Analyzers to treat `record`, `record class`, and `class` the same way. Likewise, it would treat `record struct` and `struct` the same way.
